### PR TITLE
[PORT] Limit how often you can use the emergency shuttle console

### PR DESCRIPTION
**Pull request by bloons3**
## About The Pull Request

implements https://github.com/tgstation/tgstation/pull/52522/commits/688406f878fae1730d37e89f70291c48f47c9cfa

and

https://github.com/tgstation/tgstation/pull/52522/commits/adec073d06417649fd3a850e54a5b601cb763bec

from https://github.com/tgstation/tgstation/pull/52522

Credit to @Jared-Fogle 

You can now only use the emergency shuttle console once per 5 seconds.

## Why It's Good For The Game
image

Prevents obnoxious spam of the console.

https://www.youtube.com/watch?v=0g3ZRS7Ev7w

## Changelog
:cl:
tweak: You can now only use the emergency shuttle console once per 5 seconds.
/:cl:

### CODE REVIEW

File: code/modules/shuttle/emergency.dm [hunk 1]
```diff
@@ -126,6 +134,11 @@
 	. = TRUE
 	process()
 
+/obj/machinery/computer/emergency_shuttle/proc/clear_recent_action(mob/user)
+	acted_recently -= user
+	if (!QDELETED(user))
+		ui_interact(user)
```
Comment (PowerfulBacon) [inline code/modules/shuttle/emergency.dm]: Would this not reopen the UI if they close it while standing next to it?

Comment (bloons3) [reply code/modules/shuttle/emergency.dm]: You are correct on that, I was able to reproduce that in game.
After the timeout, the UI would re-open. Should be fine to remove it.

I'm unsure if there needs to be the call to "ui_interact(user)" in the main "ui_act" proc either.

Comment (PowerfulBacon) [reply code/modules/shuttle/emergency.dm]: Does the UI not auto-update: It has a timer on it?

File: code/modules/shuttle/emergency.dm [hunk 2]
```diff
@@ -107,6 +112,9 @@
 		if(repeal)
 			minor_announce("Early launch authorization revoked, [remaining] authorizations needed.")
 			log_game("Early launch authorization revoked, [remaining] authorizations needed.")
+	acted_recently += user
+	//if(!QDELETED(user))
+	//	ui_interact(user)
```
Comment (PowerfulBacon) [inline code/modules/shuttle/emergency.dm]:
```suggestion
```

File: code/modules/shuttle/emergency.dm (hunk 1 above)
Comment (bloons3) [reply code/modules/shuttle/emergency.dm]: UI does auto-update without the calls

**Review by PowerfulBacon — APPROVED** [review]

**Comment by bloons3** [comment]
There's one annoyance with this, in that you can lock yourself out for the 5 second period by clicking revoke when you have no auth to revoke, or clicking auth when you have already given your auth.

**Pull request merged**